### PR TITLE
feat: history component for item

### LINKED ISSUE #3494: History component for Item
It would be nice to extract all item history information in Basilisk.

The component could be, by default, folded and put under the offers component.

It should come with a filter for events, and don't forget to show their royalties events as well. 

image

### ref
- #3349

### PR DESCRIPTION
**Pull request by KngZhi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3494
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer?target=DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

File: components/bsx/Gallery/Item/GalleryItem.vue [hunk 1]
```diff
@@ -263,6 +273,17 @@ export default class GalleryItem extends mixins(
...
 
+  private async fetchEvents() {
+    const result = await this.$apollo.query({
+      query: itemEvents,
+      client: this.urlPrefix,
+      variables: {
+        id: createTokenId(this.collectionId, this.id),
+      },
+    })
+    this.events = [...this.events, ...result.data.events]
```
Comment (Jarsen136) [inline components/bsx/Gallery/Item/GalleryItem.vue:284]: why we need this `[...this.events, ...result.data.events]` as there is no pagination or something.

How about `this.events = [...result.data.events]` ?

**Review by Jarsen136 — COMMENTED** [review]
small stuff

File: components/bsx/Gallery/Item/GalleryItem.vue (hunk 1 above)
Comment (KngZhi) [reply components/bsx/Gallery/Item/GalleryItem.vue:284]: Yeah, you are right. There is no need to concat.

**Comment by Jarsen136** [comment]
code LGTM

**Review by JustLuuuu — COMMENTED** [review]
wfm

**Review by petersopko — APPROVED** [review]
wfm

**Review by Jarsen136 — APPROVED** [review]

**Review by JustLuuuu — APPROVED** [review]

**Comment by Jarsen136** [comment]
pay 150

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $150 @ 59.77 USD/KSM ~ 2.51 $KSM
🧗 [DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc](https://kusama.subscan.io/account/DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)
🔗 [0xea0c00c1a2641b1d3adb7693020e9b529df5dc307dab086d08aa2f2b5b7ef6cb](https://kusama.subscan.io/extrinsic/0xea0c00c1a2641b1d3adb7693020e9b529df5dc307dab086d08aa2f2b5b7ef6cb)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by KngZhi** [comment]
@yangwao Something went wrong

image

**Comment by yangwao** [comment]
> @yangwao Something went wrong

Just refiled Guild Rewards fund:)

Screenshot 2022-07-21 at 12 42 10

**Comment by yangwao** [comment]
pay 150 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $150 @ 59.41 USD/KSM ~ 2.525 $KSM
🧗 [DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc](https://kusama.subscan.io/account/DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)
🔗 [0x890a5d56ccdaed4db8e68723ecc4ee004d335341cbbc613ad6122b9476c74457](https://kusama.subscan.io/extrinsic/0x890a5d56ccdaed4db8e68723ecc4ee004d335341cbbc613ad6122b9476c74457)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**